File: frontend/src/services/mappingRuleService.ts
import axios from 'axios';
import { MappingRule, PageResponse } from '../types';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

export const mappingRuleService = {
  getAllMappingRules: async (
    page: number = 0,
    size: number = 10,
    sortBy: string = 'name',
    direction: string = 'asc'
  ): Promise<PageResponse<MappingRule>> => {
    const params = new URLSearchParams();
    params.append('page', page.toString());
    params.append('size', size.toString());
    params.append('sortBy', sortBy);
    params.append('direction', direction);

    const response = await axios.get<PageResponse<MappingRule>>(`${API_URL}/mapping-rules?${params.toString()}`);
    return response.data;
  },

  getMappingRuleById: async (id: number): Promise<MappingRule> => {
    const response = await axios.get<MappingRule>(`${API_URL}/mapping-rules/${id}`);
    return response.data;
  },

  createMappingRule: async (mappingRuleData: Omit<MappingRule, 'id'>): Promise<MappingRule> => {
    const response = await axios.post<MappingRule>(`${API_URL}/mapping-rules`, mappingRuleData);
    return response.data;
  },

  updateMappingRule: async (id: number, mappingRuleData: Partial<MappingRule>): Promise<MappingRule> => {
    const response = await axios.put<MappingRule>(`${API_URL}/mapping-rules/${id}`, mappingRuleData);
    return response.data;
  },

  deleteMappingRule: async (id: number): Promise<void> => {
    await axios.delete(`${API_URL}/mapping-rules/${id}`);
  },

  getMappingRulesByInterface: async (
    interfaceId: number,
    page: number = 0,
    size: number = 10,
    sortBy: string = 'name',
    direction: string = 'asc'
  ): Promise<PageResponse<MappingRule>> => {
    const params = new URLSearchParams();
    params.append('page', page.toString());
    params.append('size', size.toString());
    params.append('sortBy', sortBy);
    params.append('direction', direction);

    const response = await axios.get<PageResponse<MappingRule>>(
      `${API_URL}/interfaces/${interfaceId}/mapping-rules?${params.toString()}`
    );
    return response.data;
  }
}; 

File: backend/src/main/java/com/xml/processor/service/interfaces/MappingRuleService.java
package com.xml.processor.service.interfaces;

import com.xml.processor.model.MappingRule;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface MappingRuleService {
    Page<MappingRule> getAllMappingRules(Pageable pageable);
    Page<MappingRule> getMappingRulesByClient(Long clientId, Pageable pageable);
    Page<MappingRule> getMappingRulesByInterface(Long interfaceId, Pageable pageable);
    Page<MappingRule> getMappingRulesByClientAndInterface(Long clientId, Long interfaceId, Pageable pageable);
    List<MappingRule> getActiveMappingRules(Long interfaceId);
    MappingRule createMappingRule(MappingRule mappingRule);
    MappingRule updateMappingRule(Long id, MappingRule mappingRule);
    void deleteMappingRule(Long id);
    Page<MappingRule> getMappingRules(int page, int size, String sortBy, String direction, String nameFilter, Boolean isActiveFilter);
    MappingRule getMappingRuleById(Long id);
    Page<MappingRule> searchMappingRules(String name, int page, int size, String sortBy, String direction);
    Page<MappingRule> getMappingRulesByStatus(boolean isActive, int page, int size, String sortBy, String direction);
    List<MappingRule> getMappingRulesByClientId(Long clientId);
    List<MappingRule> findByTableNameAndClient_Id(String tableName, Long clientId);
    void deleteByClient_IdAndTableName(Long clientId, String tableName);
    void saveMappingConfiguration(List<MappingRule> rules);
} 